client/signup/config/steps-pure.js

```diff
@@ -160,13 +161,15 @@ export function generateSteps( {
 				'marketing_price_group',
 				'plans_reorder_abtest_variation',
 				'allowUnauthenticated',
+				'user_id',
 			],
 			optionalDependencies: [
 				'bearer_token',
 				'username',
 				'marketing_price_group',
 				'plans_reorder_abtest_variation',
 				'allowUnauthenticated',
+				'user_id',
```

Comment:
@p-jackson I've applied your patch, thank you! I was wondering why user_id is specified as an optional dependency in the `user-new` flow, but not in the others?

Reply:
This step looks like it was added for this project pau2Xa-1hW-p2
tl;dr it looks like the plan was to allow users through the flow without signing up, hence why `bearer_token` and `username` are optional. I 

The thread says the project's on hold, and it's possible we may be able to remove this whole flow. But in the meantime, I think it's fair for `user_id` to be optional whenever `username` is optional.

Reply:
I just tested `/start/onboarding-registrationless` on this branch (which is the only flow that uses this step), and it behaves exactly the same as it does in production.